Add readme

tvrg opened the pull request:
@ArLau @NerethatAnthanael Does that help for the handover? Could you please try to compile and run noflake locally?

@jmattheis Do you have any additions or corrections?

tvrg committed:
Add readme

jmattheis commented on README.md:
```diff
@@ -1,3 +1,82 @@
+
+and use it with `go generate`. Make sure that your `$PATH` includes the go
+binary directory which is either `$(go env GOBIN)` or `$(go env GOPATH)/bin`
+(usually `~/go/bin`).
```
This could be simplified if the ``//go:generate`` statement here
https://github.com/FACT-Finder/noflake/blob/de8758fd3ca5735b83de8f1031af917965081263/main.go#L16

would use `go run` with the full package path
```go
//go:generate go run github.com/deepmap/oapi-codegen/cmd/oapi-codegen@v1.11.0 -config ./openapi-gen.yml openapi.yaml
```
Then you can ignore the `$PATH` and also don't have to install another binary.

tvrg replied:
Fixed!

jmattheis reviewed (commented):
looks good!

tvrg commented on README.md:
````diff
@@ -1,3 +1,82 @@
+1. Run Noflake:
+   ```bash
+    docker run \
+        -v $PWD/noflake.sqlite3:/opt/noflake/noflake.sqlite3 \
````
Doesn't work if you noflake.sqlite3 doesn't exist yet. Docker then creates an empty directory.

tvrg replied:
Fixed! I'll release 0.1.0 and update the ansible deployment script after merging.

tvrg committed:
Move sqlite database to /var/lib/noflake

Having the noflake binary and the database file both located in the same
/opt/noflake directory is suboptimal.

Directly mounting the database file as in

    docker run -v /data/noflake.sqlite3:/opt/noflake/noflake.sqlite3 ...

isn't a good option when initially /data/noflake.sqlite3 doesn't exist
because then docker creates an empty directory at /data/noflake.sqlite3.

Instead you want to mount the directory containing the database file
using

    docker run -v /data:/opt/noflake ...

But this fails because it also overrides the noflake binary at
/opt/noflake/noflake. The solution is to put the database file into a
separate directory, in our case /var/lib/noflake.

tvrg committed:
Simplify go generate usage

tvrg committed:
Describe how to build the docker image
